frontend/src/js/controllers/updateProfileCtrl.js

```diff
@@ -0,0 +1,98 @@
+// Invoking IIFE for teams
+
+(function() {
+
+    'use strict';
+
+    angular
+        .module('evalai')
+        .controller('updateProfileCtrl', updateProfileCtrl);
+
+    updateProfileCtrl.$inject = ['utilities', '$state', '$http', '$rootScope'];
+
+    function updateProfileCtrl(utilities, $state, $http, $rootScope) {
+        /* jshint validthis: true */
+        var vm = this;
+        var userKey = utilities.getData('userKey');
+        vm.wrnMsg = {};
+        vm.isValid = {};
+        vm.user = {};
+        vm.isFormError = false;
+
+        vm.updateprofileContainer = angular.element('.update-profile-card');
+
+        vm.startLoader = function(msg) {
+            $rootScope.isLoader = true;
+            $rootScope.loaderTitle = msg;
+            vm.updateprofileContainer.addClass('low-screen');
+        };
+
+        // stop loader
+        vm.stopLoader = function() {
+            $rootScope.isLoader = false;
+            $rootScope.loaderTitle = '';
+            vm.updateprofileContainer.removeClass('low-screen');
+        };
+        // function to update Profile
+        vm.updateProfile = function(resetconfirmFormValid) {
+            if (resetconfirmFormValid) {
+
+                vm.startLoader("Updating Your Profile");
+                var parameters = {};
+                parameters.url = 'auth/user/';
+                parameters.method = 'PUT';
+                parameters.data = {
+                    "username": vm.user.username,
+                    "first_name": vm.user.first_name,
+                    "last_name": vm.user.last_name,
+                    "uid": $state.params.user_id,
```

Comment:
there is no parameters in url called `user_id`. No need of it. Remove this line.

PS: `$state.params` is used when some data like user id or any token is present in the url (i.e password reset email link has parameters like, user_token etc.)